pkg/jobs/hive/hive.go

```diff
@@ -266,175 +267,39 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
 
 	klog.V(2).Info("Check if managedclusterview exists " + clusterName)
 
-	managedclusterview := &managedclusterviewv1beta1.ManagedClusterView{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      clusterName,
-			Namespace: clusterName,
-			Labels: map[string]string{
-				MCVUpgradeLabel: clusterName,
-			},
-		},
-		Spec: managedclusterviewv1beta1.ViewSpec{
-			Scope: managedclusterviewv1beta1.ViewScope{
-				Group:     "config.openshift.io",
-				Kind:      "ClusterVersion",
-				Name:      "version",
-				Namespace: "",
-				Version:   "v1",
-			},
-		},
-	}
-
-	mcview := managedclusterviewv1beta1.ManagedClusterView{}
-	if err := client.Get(context.TODO(), types.NamespacedName{
-		Namespace: clusterName,
-		Name:      clusterName,
-	}, &mcview); err != nil {
+	successful := false
+	for i := 1; i <= retries && !successful; i++ {
```

Comment:
I don't see any toggles for this new behaviour. 

Is this changing the existing behaviour so before it fails a lot sooner now it always retries 10 times? 

CC @rokej @xiangjingli

Reply:
@mikeshng It's further down in this line
![image](https://github.com/stolostron/cluster-curator-controller/assets/38960034/4883b1e4-d327-471c-a56c-03072ee4042a)

Before, we just try to update the clusterversion once but with the new change I added a loop to retry up to 10 times and if one of the attempts succeed it will break out of the loop.